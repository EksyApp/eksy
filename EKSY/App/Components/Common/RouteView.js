import React, {Component} from 'react'
import {View, StyleSheet} from 'react-native'
import RouteMap from './RouteMap'
import Label from './Label'
import TextArea from './TextArea'
import Divider from './Divider'
import MarkerCardList from './MarkerCardList'
import PropTypes from 'prop-types'
import {RegionShape, RouteShape} from "../../Utils/PropTypeShapes";

export default class RouteView extends Component {
	
	render() {
		return(
				<View>
					<View style={styles.mapContainer}>
						<RouteMap
								style={styles.map}
								initialRegion={this.props.initialRegion}
								markers={this.props.route.markers}
						/>
					</View>
					<Label>{this.props.route.title}</Label>
					<TextArea>{this.props.route.text}</TextArea>
					<Divider/>
					<MarkerCardList data={this.props.route.markers} onPress={this.props.onMarkerClick} style={styles.card}/>
				</View>
		)
	}
	
}

const styles = StyleSheet.create({
	
	mapContainer: {
		height: 300,
		width: '100%',
		marginBottom: 20
	},
	
	map: {
		width: '100%',
		height: '100%'
	},
	
	card: {
		width: '95%',
		alignSelf: 'center'
	}
})

RouteView.propTypes = {
	initialRegion: RegionShape,
	route: RouteShape,
	onMarkerClick: PropTypes.func,
}