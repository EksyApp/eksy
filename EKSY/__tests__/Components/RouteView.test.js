import React from 'react';
import renderer from 'react-test-renderer';
import RouteView from "../../App/Components/Common/RouteView";

test('RouteViewComponent renders correctly', () => {
	const tree = renderer.create(
			<RouteView
					initialRegion = {{latitude: 1, longitude: 1, latitudeDelta: 1, longitudeDelta: 1}}
					route = {{markers: [{latitude: 1, longitude: 1}]}}
			/>
	).toJSON();
	expect(tree).toMatchSnapshot();
});