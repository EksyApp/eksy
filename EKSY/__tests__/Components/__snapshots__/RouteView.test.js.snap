// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteViewComponent renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "height": 300,
        "marginBottom": 20,
        "width": "100%",
      }
    }
  >
    <MapView
      initialRegion={
        Object {
          "latitude": 1,
          "latitudeDelta": 1,
          "longitude": 1,
          "longitudeDelta": 1,
        }
      }
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <Marker
        coordinate={
          Object {
            "latitude": 1,
            "longitude": 1,
          }
        }
      />
      <Polyline
        coordinates={
          Array [
            Object {
              "latitude": 1,
              "longitude": 1,
            },
          ]
        }
      />
    </MapView>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "rgb(101,114,122)",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        },
        undefined,
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "rgb(89,85,78)",
          "fontSize": 18,
          "margin": 10,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#e1e8ee",
          "height": 1,
        },
        Array [
          Object {
            "backgroundColor": "rgb(112,112,112)",
            "height": 1,
            "margin": 3,
            "marginLeft": 10,
            "marginRight": 10,
          },
          undefined,
        ],
      ]
    }
  />
  <RCTScrollView
    dataSource={
      ListViewDataSource {
        "items": 1,
      }
    }
    enableEmptySections={true}
    renderRow={[Function]}
    renderScrollComponent={[Function]}
  >
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "rgb(226,227,217)",
              "borderBottomWidth": 0,
              "borderColor": "#ddd",
              "borderRadius": 2,
              "borderWidth": 1,
              "elevation": 1,
              "marginLeft": 5,
              "marginTop": 10,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 2,
            },
            Array [
              Object {
                "height": 200,
                "marginLeft": 0,
                "width": "100%",
              },
              Object {
                "width": 740,
              },
              Object {
                "alignSelf": "center",
                "width": "95%",
              },
            ],
          ]
        }
        testID={undefined}
      >
        <View
          style={
            Array [
              Object {
                "borderBottomWidth": 1,
                "borderColor": "rgb(112,112,112)",
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "padding": 5,
                "position": "relative",
              },
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "overflow": "scroll",
                "width": "100%",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "rgb(101,114,122)",
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                Object {
                  "marginTop": 10,
                },
              ]
            }
          />
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "rgb(89,85,78)",
                "fontSize": 18,
                "margin": 10,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
